### PR TITLE
Add MoonPay on-ramp support. Add CUSD and CEUR support for Transak

### PR DESCRIPTION
**Pull request by wachunei**
This PR adds MoonPay support

**⚠️ Keys** 
This PR needs a pair of keys added to `.js.env`, please ping me for values.
```js
export MOONPAY_API_KEY_STAGING= // <sandbox key>
export MOONPAY_API_KEY_PRODUCTION= // <prod key>
```

### CODE REVIEW

**Review by gantunesr — COMMENTED** [review]
@wachunei do you have capacity to refactor the new components from JS to TS?

**Comment by wachunei** [comment]
> @wachunei do you have capacity to refactor the new components from JS to TS?

No new components are introduced, all are refactors from current ones. Which ones do you mean?

**Comment by gantunesr** [comment]
> No new components are introduced, all are refactors from current ones. Which ones do you mean?

Okay, I didn't realized that those were refactor components, I thought `MoonPayWebView` and `MoonPayPaymentMethod` were a new one.

Is `orderProcessor/moonpay.js` also a refactor or a new file?

**Comment by wachunei** [comment]
`MoonPayWebView` and `MoonPayPaymentMethod`  are new files but based on the same existing components. The `orderProcessor/moonpay.js` file is new but very similar to previous providers processor files.

File: app/components/UI/FiatOrders/PaymentMethodSelector/moonpay.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+const styles = StyleSheet.create({
+	title: {
+		flex: 1,
+		flexWrap: 'wrap',
+	},
+	logo: {
+		marginVertical: 5,
+		aspectRatio: 137.18 / 25,
+		// width: Device.isIphone5() ? 109.75 : 137.18,
+		// height: Device.isIphone5() ? 20 : 25,
```
Comment (wachunei) [inline app/components/UI/FiatOrders/PaymentMethodSelector/moonpay.js:21]: Remove this
```suggestion
```

File: app/components/UI/FiatOrders/orderProcessor/moonpay.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+const getTransactionAmount = (transaction) =>
+	transaction ? transaction.baseCurrencyAmount + getTransactionFee(transaction) : 0;
+
+//* Constants
+
+const { MOONPAY_URL, MOONPAY_URL_STAGING, MOONPAY_API_URL_PRODUCTION, MOONPAY_REDIRECT_URL } = AppConstants.FIAT_ORDERS;
+
+const isDevelopment = process.env.NODE_ENV !== 'production';
+
+const MOONPAY_API_BASE_URL = isDevelopment ? MOONPAY_API_URL_PRODUCTION : MOONPAY_API_URL_PRODUCTION;
```
Comment (wachunei) [inline app/components/UI/FiatOrders/orderProcessor/moonpay.js:122]: Add a comment to explain this line, which is misleading
```suggestion
// MoonPay uses the same API URL for orders created in their sandbox and production environments
const MOONPAY_API_BASE_URL = isDevelopment ? MOONPAY_API_URL_PRODUCTION : MOONPAY_API_URL_PRODUCTION;
```

**Review by Fatxx — APPROVED** [review]
LGTM

**Pull request merged**